flink-external-resource/flink-external-resource-gpu/src/main/resources/gpu-discovery.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+if [ $# -lt 1 ]; then
+  echo "Usage: ./gpu-discovery-script.sh gpu-amount [--privilege] [--check-dead] [--assign-file filePath]"
+  exit 1
+fi
+
+
+AMOUNT=$1
+shift
+ASSIGNMENT_FILE="/var/tmp/flink-gpu-assignment"
+
+while [[ $# -ge 1 ]]
+do
+key="$1"
+  case $key in
+    --privilege)
+    PRIVILEGE_OPTION="privilege"
+    ;;
+    --check-dead)
+    CHECK_DEAD_PROCESS="check"
+    ;;
+    --assign-file)
+    ASSIGNMENT_FILE="$2"
+    ;;
+    *)
+    # unknown option
+    ;;
+  esac
+  shift
```

Comment:
Usually, I would suggest to avoid `shift` for better readability.
In this particular case, I think it probably make sense to use shift because we do not require certain order of the arguments. However, I would suggest to `shift` inside each case branch. It would be better to have an explicit `shift 2` for `--assign-file` which takes two arguments, rather than relying on ignoring unknown tokens.

Reply:
Not sure how shift hurt the readability. I just miss adding `shift` to `--assign-file` branch here.